lineapy/graph_reader/apis.py

```diff
@@ -100,22 +100,33 @@ def _get_value_path(
     # Note that I removed the @properties becuase they were not working
     # well with the lru_cache
     @lru_cache(maxsize=None)
-    def _get_subgraph(self) -> Graph:
+    def _get_subgraph(self, keep_lineapy_save: bool = False) -> Graph:
         """
-        Return the slice subgraph for the artifact
+        Return the slice subgraph for the artifact.
+
+        :param keep_lineapy_save: Wheather to retain ``lineapy.save()`` in code slice.
+                Defaults to ``False``.
+
         """
-        return get_slice_graph(self._get_graph(), [self._node_id])
+        return get_slice_graph(
+            self._get_graph(), [self._node_id], keep_lineapy_save
+        )
 
     @lru_cache(maxsize=None)
-    def get_code(self, use_lineapy_serialization=True) -> str:
+    def get_code(
+        self,
+        use_lineapy_serialization: bool = True,
+        keep_lineapy_save: bool = False,
+    ) -> str:
         """
         Return the slices code for the artifact
 
-        :param use_lineapy_serialization: If True, will use the lineapy serialization to get the code.
-            We will hide the serialization and the value pickler irrespective of the value type.
-
-            If False, will use remove all the lineapy references and instead use the underlying serializer directly.
-            Currently, we use the native `pickle` serializer.
+        :param use_lineapy_serialization: If ``True``, will use the lineapy serialization to get the code.
+                We will hide the serialization and the value pickler irrespective of the value type.
+                If ``False``, will use remove all the lineapy references and instead use the underlying serializer directly.
+                Currently, we use the native ``pickle`` serializer.
+        :param keep_lineapy_save: Wheather to retain ``lineapy.save()`` in code slice.
```

Comment:
Also, including the `.save` for values make sense because there is a saving of values, however for side-effects, it's kinda weird to include e.g., `lineapy.save(lineapy.file_system)` since it doesn't do anything except create a new version of the artifact (it doesn't even have the serialization logic). I wonder how we can make this less confusing.

Reply:
I guess you meant the case of setting `use_lineapy_serialization=False` and `keep_lineapy_save=True`, which seems contradictory on surface? Actually, `keep_lineapy_save` is functionally closer to "keep_artifact_save", meaning it decides whether to retain statements relating to artifact storage (whether that is of lineapy or pickle form). So, given

```python
import lineapy

n = 10
art = lineapy.save(n, "n")
```

we get:

```python
print(art.get_code(use_lineapy_serialization=False, keep_lineapy_save=True))
"""
import pickle

n = 10
art = pickle.dump(n, open("/Users/sangyoonpark/.linea/linea_pickles/FdzwmFS", "wb"))
"""
```

```python
print(art.get_code(use_lineapy_serialization=False, keep_lineapy_save=False))
"""
n = 10
"""
```

```python
print(art.get_code(use_lineapy_serialization=True, keep_lineapy_save=True))
"""
import lineapy

n = 10
art = lineapy.save(n, "n")
"""
```

```python
print(art.get_code(use_lineapy_serialization=True, keep_lineapy_save=False))
"""
n = 10
"""
```

I agree these two Boolean parameters might be confusing though, so we can discuss UX further as you suggested!

Reply:
Yeah, that's a good point. Actually, I noticed that `lineapy.save(lineapy.file_system)` is never retained even if we set `keep_lineapy_save=True`. So, given 

```python
import lineapy
import pandas as pd

df = pd.read_csv("https://raw.githubusercontent.com/LineaLabs/lineapy/main/examples/tutorials/data/iris.csv")
df["test"] = "test"
df.to_csv("new_iris.csv")
art = lineapy.save(lineapy.file_system, "file_system")
```

we get:

```python
print(art.get_code(keep_lineapy_save=True))
"""
import pandas as pd

df = pd.read_csv(
    "https://raw.githubusercontent.com/LineaLabs/lineapy/main/examples/tutorials/data/iris.csv"
)
df["test"] = "test"
df.to_csv("new_iris.csv")
"""
```

Again, I agree that we need to discuss UX here further -- for different scenarios.